base_layer/core/src/validation/transaction_validators.rs

```diff
@@ -129,17 +129,15 @@ fn verify_not_stxos<B: BlockchainBackend>(tx: &Transaction, db: &B) -> Result<()
                 );
```

Comment:
Github does not let me comment on the above lines, but should we not replace this piece of code above to be the same as the block validation. The block validation now uses `db.is_unspent_output(&input.commitment)? {` while this still uses the deletedbitmap.

Reply:
I did start to change this, but changed it back. tbh I wasn't too sure. 
Thinking about it now, I think we need the old check to test that the output was used as is (i.e. by output hash) and is not spent AND check that the commitment is not already in the current UTXO set AND the commitment is not already in the UTXO set.

Reply:
On seconds thoughts, I think just using is_unspent_output is fine.

These are the spends we want to prevent:

```
Spending to same commitment in current UTXO set
U1 -> U2
U3 -> U2
-------------
Double spending output, even if has different output features (^)
U1 -> U2
U1^ -> U3
-------------
Spent to output with same commitment
U1 -> U1^
```

but not sure about these

```
(A)
U1 -> U2
U2 -> U1
-------------
(B)
U1 -> U2
U2 -> U1^
```
Currently (A) will fail validation because the hash exists in the MMR. It would also fail on the db level because we currently have a unique index on all outputs (`txos_hash_to_index_db`) but this is another possible point of dicussion.
(B) will work (but only as long as you change output features each time)

Reply:
Updated - The wallet uses the transaction validator to tell it (via rpc) if a transaction was mined or not, so we still need to query the output hash (fetch_output).

Reply:
awesome, I hate the fact we still have this duplicated code. But we can't have the transactions and block validators running different rules. Mempool uses transactions validation to decide on the transactions to include on the block, while base nodes use the block validation to decide on blocks. If a tx passes tx_validation and fails block validation, its a problem.

Reply:
Yeah not sure we're going to get around that for the moment - mostly because the mempool uses the transaction validation to tell the wallet things it needs to know (e.g was a transaction mined) and so needs to do extra things to determine this that is not necessary if just checking the "spendability" of a UTXO (i.e. what block validator cares about) - so for now I'll keep the duplicate code. They do actually have subtly different roles - so I don't think this is too bad if we make sure our tests check that the transaction validator never let's through anything that block validator will reject.